### PR TITLE
Add Traefik prefix to the KV key

### LINKED ISSUE #2394: KV does not respect "prefix" when testing KV store connection
### Do you want to request a *feature* or report a *bug*?
Report a bug.

### What did you do?
Used an ACL in Consul 1.0.0 to restrict Traefik to read/write access "traefik" key/folder. [consul] "prefix" is set to "traefik".

### What did you expect to see?
I expected Traefik to only need write access to the key/folder specified in [consul] "prefix" .

### What did you see instead?
Traefik failed to start with an error message in the logs of "KV connection error: Failed to test KV store connection: Unexpected response code 403."

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.4.2
Codename:     roquefort
Go version:   go1.9.2
Built:        2017-11-02_01:47:46PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
Using consul 1.0.0 in a separate docker stack, with a shared network between traefik and consul.

traefik docker-compose.yml snippet:
```yaml
version: '3.4'
services:
  traefik:
    image: traefik
    command: -c /dev/null --web --docker --docker.domain=docker.localhost --logLevel=DEBUG --consul --consul.endpoint=consul_server:8500
```

Working ACL (as a workaround):
```
key "traefik" {
  policy = "write"
},
key "qmslkjdfmqlskdjfmqlksjazçueznbvbwzlkajzebvkwjdcqmlsfj" {
  policy = "write"
}
```
Working ACL (full root access; not desirable):
```
key "" {
  policy = "write"
}
```
Non-Working and (desirable) ACL:
```
key "traefik" {
  policy = "write"
}
```
Using an ACL with full read/write access or read/write access to the specific test key (found in the source code) allows traefik to fully work as expected.

This bug resides on line 105 of provider/kv/kv.go https://github.com/containous/traefik/blob/7192aa86b5264e014ceabceb6367d8e6ce9a2fe6/provider/kv/kv.go#L105

Instead of writing and checking the test key within the "prefix" folder, traefik attempts to write and read it from the root. This is not desirable from a security standpoint, is outside of the expected behavior, and is not documented.

I don't have a build environment up and running yet, but may contribute a pull to fix this bug at a later time.

### PR DESCRIPTION
**Pull request by nmengin**
### What does this PR do?

Add the prefix `Treafik` to the key used to check the connection to the KV stores.

### Motivation

Fixes #2394 

### More

- [x] Updated documentation

### CODE REVIEW

File: docs/user-guide/kv-config.md [hunk 1]
```diff
@@ -148,6 +148,18 @@ This variable must be initialized with the ACL token value.
 
 If Traefik is launched into a Docker container, the variable `CONSUL_HTTP_TOKEN` can be initialized with the `-e` Docker option : `-e "CONSUL_HTTP_TOKEN=[consul-acl-token-value]"`
 
+!!! note
+    If a Consul ACL is used to restrict Træfik read/write access, the following configuration is needed :
+
+```
```
Comment (ldez) [inline docs/user-guide/kv-config.md]: Wrong render: indent the code block.

File: provider/kv/kv.go [hunk 2]
```diff
@@ -34,7 +37,7 @@ type Provider struct {
 func (p *Provider) CreateStore() (store.Store, error) {
 	storeConfig := &store.Config{
 		ConnectionTimeout: 30 * time.Second,
-		Bucket:            "traefik",
+		Bucket:            traefikKVPrefix,
```
Comment (ldez) [inline provider/kv/kv.go]: I think this is not a prefix but a bucket name.

File: provider/kv/kv.go [hunk 3]
```diff
@@ -102,7 +105,7 @@ func (p *Provider) watchKv(configurationChan chan<- types.ConfigMessage, prefix
 func (p *Provider) Provide(configurationChan chan<- types.ConfigMessage, pool *safe.Pool, constraints types.Constraints) error {
 	p.Constraints = append(p.Constraints, constraints...)
 	operation := func() error {
-		if _, err := p.kvclient.Exists("qmslkjdfmqlskdjfmqlksjazçueznbvbwzlkajzebvkwjdcqmlsfj"); err != nil {
+		if _, err := p.kvclient.Exists(traefikKVPrefix + "/qmslkjdfmqlskdjfmqlksjazçueznbvbwzlkajzebvkwjdcqmlsfj"); err != nil {
```
Comment (ldez) [inline provider/kv/kv.go]: `p.prefix` or  bucket name ?

```go
Prefix                string           `description:"Prefix used for KV store" export:"true"`
```

File: docs/user-guide/kv-config.md (hunk 1 above)
Comment (nmengin) [reply docs/user-guide/kv-config.md]: Code indented under this block.

**Pull request merged**